create_near_neighbor_ids.py

```diff
@@ -0,0 +1,51 @@
+# coding: utf-8
+
+"""
+Create a file near_neighbor_ids.fits.gz which contains a single column
+of AGASC IDs corresponding to all stars in AGASC 1.7 that are within
+60 arcsec of a candidate guide or acq star.
+
+This takes a while to run and should be done on a computer with copy
+of AGASC 1.7 on a local (fast) drive.
+
+Usage::
+
+  $ python make_near_neighbor_ids.py
+"""
+import tables
+import os
+from pathlib import Path
+
+import agasc
+from astropy.table import Table
+
+agasc1p7 = str(Path(os.environ['SKA'], 'data', 'agasc', 'agasc1p7.h5'))
+
+h5 = tables.open_file(agasc1p7)
+stars = h5.root.data[:]
+h5.close()
+
+ok = ((stars['CLASS'] == 0) &
+      (stars['MAG_ACA'] < 11.0) &
+      (stars['ASPQ1'] < 50) &  # Less than 2.5 arcsec from nearby star
+      (stars['ASPQ1'] > 0) &
```

Comment:
Has this been checked?  Meaning is it true that all candidates stars in the AGASC that have a neighbor within 60 arcsecs also have nonzero ASPQ1?  Seems fine but if it is true, we also don't need the filter here, right?  Because removing the filter will add no extra neighbor stars.

Reply:
What I believe, and the basis for operations since AGASC 1.5, is that no star with ASPQ1 = 0 can have any near neighbor stars that impact tracking or acquisition.  Not sure if we have to reevaluate in light of impact to background pixels, but I think this is unlikely to be a problem in reality.

So skipping ASPQ1 = 0 stars makes the create_near_neighbors script run *way* faster.  Those are the majority of stars.  With this filter in place, it takes about an hour on my laptop, much longer on kadi.  Without, probably 85 hours IIRC.

Reply:
I just thought the whole point of this exercise was that you were reevaluating in light of the impact of background pixels.